Library/Homebrew/dev-cmd/bump-formula-pr.rb

```diff
@@ -15,6 +15,39 @@
 require "formula"
 
 module Homebrew
+  def inreplace_pairs(formula, replacement_pairs)
+    if ARGV.dry_run?
+      contents = formula.path.open("r") { |f| Formulary.set_encoding(f).read }
+      contents.extend(StringInreplaceExtension)
+      replacement_pairs.each do |old, new|
+        ohai "replace \"#{old}\" with \"#{new}\"" unless ARGV.flag?("--quiet")
+        contents.gsub!(old, new)
+      end
+      if contents.errors.any?
+        raise Utils::InreplaceError, formula.path => contents.errors
+      end
+      contents
+    else
+      Utils::Inreplace.inreplace(formula.path) do |s|
+        replacement_pairs.each do |old, new|
+          ohai "replace \"#{old}\" with \"#{new}\"" unless ARGV.flag?("--quiet")
+          s.gsub!(old, new)
+        end
+      end
+      formula.path.open("r") { |f| Formulary.set_encoding(f).read }
+    end
+  end
```

Comment:
The confidence in knowing that the non-dry-run path will automatically get whatever changes are made to inreplace in the future. But this duplication is exactly why I was saying I wish inreplace itself had a dry-run mode since this whole function feels like a big hack around that deficiency,